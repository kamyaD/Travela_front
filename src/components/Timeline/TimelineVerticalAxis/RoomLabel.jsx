/* eslint-disable jsx-a11y/click-events-have-key-events */
import React, {PureComponent} from 'react';
import {PropTypes} from 'prop-types';
import MaintainceForm from '../../Forms/MaintainanceForm';
import tick from '../../../images/Tick/tick.svg';
import Modal from '../../modal/Modal';

class RoomLabel extends PureComponent {
  state = {
    showMarkUnavailable: false
  };

  toggleMarkUnavailable = () => {
    this.setState(prevState => ({
      showMarkUnavailable: !prevState.showMarkUnavailable
    }));
  }

  renderMainteinanceForm(type){
    const {closeModal, name, id, status, shouldOpen, addmaintenanceRecord,
      updateMaintenanceRecord, modalType, timelineDateRange, guestHouseId} = this.props;
    return(
      <Modal
        closeModal={closeModal}
        width="480px"
        customModalStyles="room-maintanance"
        visibility={shouldOpen && type === modalType ? 'visible' : 'invisible'}
        title={`Mark ${name} Unavailable`}
      >
        <MaintainceForm
          status={status}
          id={id}
          closeModal={closeModal}
          addmaintenanceRecord={addmaintenanceRecord}
          timelineDateRange={timelineDateRange}
          guestHouseId={guestHouseId}
          updateMaintenanceRecord={updateMaintenanceRecord}
        />
      </Modal>
    );
  }

  render() {
    const {showMarkUnavailable} = this.state;
    const {name, id, openModal, status} = this.props;
    const visibility = showMarkUnavailable ? 'is-visible' : 'is-hidden';
    const statusClass = status ? 'fault' : 'fine';
    return (
      <div className="room-name item-row">
        <div className="name">{name}</div>
        <div
          className="ellipsis"
          tabIndex="0"
          role="button"
          onClick={this.toggleMarkUnavailable}
        >
          &hellip;
          <div 
            className={`mark-unavailable ${visibility}`}
            role="button"
            tabIndex="0"
            onClick={() => !status && openModal(true,  `${name}-${id}`)}
          >
            <div className={`container_room_${statusClass}`} />
            <span>Unavailable</span>
          </div>
        </div>
        {this.renderMainteinanceForm(`${name}-${id}`)}
      </div>
    );
  }
}

RoomLabel.propTypes = {
  closeModal: PropTypes.func.isRequired,
  addmaintenanceRecord: PropTypes.func.isRequired,
  updateMaintenanceRecord: PropTypes.func.isRequired,
  shouldOpen: PropTypes.bool.isRequired,
  status: PropTypes.bool.isRequired,
  timelineDateRange: PropTypes.array.isRequired,
  guestHouseId: PropTypes.string.isRequired,
  modalType: PropTypes.string,
  openModal: PropTypes.func.isRequired,
  name: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired,
};
RoomLabel.defaultProps = {
  modalType: null
};

export default RoomLabel;
