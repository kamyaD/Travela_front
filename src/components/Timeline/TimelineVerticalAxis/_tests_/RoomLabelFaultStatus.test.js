import React from 'react';
import { shallow } from 'enzyme';
import sinon from 'sinon';
import RoomLabel from '../RoomLabel';

describe('<RoomLabel />', ()=> {
  const props = {
    updateRoomState: jest.fn(),
    openModal: jest.fn(),
    closeModal: jest.fn(),
    addmaintenanceRecord: jest.fn(),
    name: 'qwetu',
    id: 'thjfsndj12',
    status: false,
    timelineDateRange: ['12/12/2018', '17/12/2018'],
    guestHouseId: 'fhdj212',
    shouldOpen: true,
    modalType: 'maintainance'
  };

  const props2 = {
    updateRoomState: jest.fn(),
    closeModal: jest.fn(),
    openModal: jest.fn(),
    addmaintenanceRecord: jest.fn(),
    name: 'qwetu',
    id: 'thjfsndj12',
    status: true,
    timelineDateRange: ['12/12/2018', '17/12/2018'],
    guestHouseId: 'fhdj212',
    shouldOpen: true,
    modalType: 'maintainance'
  };
  window.date = {
    format: jest.fn(),
  };

  const shallowWrapper = shallow(<RoomLabel {...props} />);

  it('should match snapshot', () => {
    const wrapper = shallow(<RoomLabel {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should not show checkbox on mount', () => {
    const wrapper = shallow(<RoomLabel {...props} />);
    const checkbox = wrapper.find('.is-visible');
    expect(checkbox.length).toBe(0);
  });

  it('should show checkbox after state changes', () => {
    const wrapper = shallow(<RoomLabel {...props} />);
    wrapper.setState({showMarkUnavailable: true });
    const checkbox = wrapper.find('.is-visible');
    expect(checkbox.length).toBe(1);
  });


  it('should show unmarked checkbox', () => {
    const wrapper = shallow(<RoomLabel {...props2} />);
    const checkbox = wrapper.find('.container_room_fault');
    expect(checkbox.length).toBe(1);
  });

  it('should show marked checkbox', () => {
    const wrapper = shallow(<RoomLabel {...props} />);
    const checkbox = wrapper.find('.container_room_fine');
    expect(checkbox.length).toBe(1);
  });

  it('calls update room status when clicked with status of true', () => {
    const shallowWrapper = shallow(<RoomLabel {...props2} />);
    const spy = sinon.spy(shallowWrapper.instance(), 'showId');
    const checkboxBlue = shallowWrapper.find('.container_room_fault');
    checkboxBlue.simulate('click');
    checkboxBlue.simulate('keyDown');
    expect(spy.calledTwice).toBe(true);
  });

  it('calls update room status when clicked with status of false', () => {
    const spy = sinon.spy(shallowWrapper.instance(), 'showId');
    const checkbox = shallowWrapper.find('.container_room_fine');
    checkbox.simulate('click');
    checkbox.simulate('keyDown');
    expect(spy.calledOnce).toBe(false);
  });

  it('changes the state of the elipses when clicked', () => {
    const wrapper = shallow(<RoomLabel {...props} />);
    const elipses = wrapper.find('.ellipsis');
    elipses.simulate('focus');
    expect(wrapper.state().showMarkUnavailable).toEqual(true);
  });
});
