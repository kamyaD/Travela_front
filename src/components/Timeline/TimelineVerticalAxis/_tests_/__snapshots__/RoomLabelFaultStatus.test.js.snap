// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RoomLabel /> should match snapshot 1`] = `
<div
  className="room-name item-row"
>
  <div>
    qwetu
  </div>
  <div
    className="ellipsis"
    onBlur={[Function]}
    onFocus={[Function]}
    role="button"
    tabIndex="0"
  >
    â€¦
    <div
      className="mark-unavailable is-hidden"
    >
      <div>
        <div
          className="container_room_fine"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
          tabIndex="0"
        />
      </div>
      <span>
        Unavailable
      </span>
    </div>
  </div>
  <Modal
    closeDeleteModal={null}
    closeModal={[Function]}
    customModalStyles=""
    customOverlayStyle=""
    modalBar={<div />}
    modalContentId=""
    modalId=""
    requestId=""
    showOverlay={true}
    title="Mark qwetu Unavailable"
    visibility="invisible"
    width="480px"
  >
    <MaintainceForm
      addmaintenanceRecord={[Function]}
      closeModal={[Function]}
      editMaintenance={Object {}}
      guestHouseId="fhdj212"
      id="thjfsndj12"
      maintenance={Object {}}
      modalType=""
      status={false}
      timelineDateRange={
        Array [
          "12/12/2018",
          "17/12/2018",
        ]
      }
      updateMaintenanceRecord={[Function]}
    />
  </Modal>
</div>
`;
